File: codechef/DEC9.py
import sys

front = []
back = []
def max_subarray(A,n):
    max_ending_here = max_so_far = A[0]
    max_ending_here1 = max_so_far1 = A[n-1]
    front.append(max_so_far)
    back.append(max_so_far1)
    x = 1
    while x < n:
        max_ending_here = max(A[x], max_ending_here + A[x])
        max_so_far = max(max_so_far, max_ending_here)
    	front.append(max_so_far)
    	max_ending_here1 = max(A[n-1-x], max_ending_here1 + A[n-1-x])
        max_so_far1 = max(max_so_far1, max_ending_here1)    	
        back.insert(0,max_so_far1) 
        x+=1   

t = input()
while t:
	n,k = map(int,sys.stdin.readline().split())
	A = map(int,sys.stdin.readline().split())
	front = []
	back = []
	max_subarray(A,n)
	res = -100000000000000000000		
	for x in range(0,n-k-1):
		res = max(front[x] + back[x+k+1] , res)
	print(res)	 
	t-=1	            
	

File: Codechef/DEC9.cpp
#include<vector>
#include<algorithm>
#include<iostream>
#include<stack>
#include<queue> 
using namespace std;
long long int front[100001];
long long int back[100001];

int maxray(long long int A[],long int n){
long long int maxhere,maxhere1;
long long int maxsofar1,maxsofar;
maxhere = A[0];
maxsofar = maxhere;
maxhere1 = A[n-1];
maxsofar1 = maxhere1;
front[0] =  maxhere;
back[n-1]  =  maxhere1;
long int x = 1;
while(x < n){
maxhere = max(A[x], maxhere + A[x]);
maxsofar = max(maxsofar, maxhere);
front[x] = maxsofar;
maxhere1 = max(A[n-1-x], maxhere1 + A[n-1-x]);
maxsofar1 = max(maxsofar1, maxhere1);    	
back[n-1-x]= maxsofar1; 
x+=1;
}
} 
int main(){
 int t;
 cin>>t;
 while(t--){

 long int n,k;
 cin>>n>>k;
 long long int a[n];
 for(long int i = 0 ; i < n ; i++) cin>>a[i];
 maxray(a,n);
 long long int res = -2000000001;
 for(long int i = 0 ; i < n-k-1;i++){
 res = max(front[i] + back[i+k+1] , res);
 }
 cout<<res<<endl;
 }
 return 0;
 }
